source/common/http/conn_manager_impl.cc

```diff
@@ -742,6 +742,16 @@ const Network::Connection* ConnectionManagerImpl::ActiveStream::connection() {
   return &connection_manager_.read_callbacks_->connection();
 }
 
+uint32_t ConnectionManagerImpl::ActiveStream::localPort() {
+  auto conn = connection();
+  ASSERT(conn != nullptr);
```

Comment:
nit: generally we don't do asserts like this since it will crash in an obvious way on the next line. I would just remove the conn variable.